From d8514a2a0572f23a1bda6cd4fe32ee3b6e805719 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 11 Dec 2013 21:10:14 +0100
Subject: [PATCH 323/327] MINOR: checks: create the agent tasks even when no
 check is configured

Till now, a configuration required at least one health check in the
whole config file to create the agent tasks. Now we start them even
if no health check is enabled.
---
 src/checks.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/src/checks.c b/src/checks.c
index 382bc59..47dd523 100644
--- a/src/checks.c
+++ b/src/checks.c
@@ -1765,19 +1765,19 @@ int start_checks() {
 				t->expire = TICK_ETERNITY;
 			}
 
-			if (!(s->check.state & CHK_ST_CONFIGURED))
-				continue;
-
-			if ((srv_getinter(&s->check) >= SRV_CHK_INTER_THRES) &&
-			    (!mininter || mininter > srv_getinter(&s->check)))
-				mininter = srv_getinter(&s->check);
-
-			if ((s->agent.state & CHK_ST_CONFIGURED) &&
-			    (srv_getinter(&s->agent) >= SRV_CHK_INTER_THRES) &&
-			    (!mininter || mininter > srv_getinter(&s->agent)))
-				mininter = srv_getinter(&s->agent);
+			if (s->check.state & CHK_ST_CONFIGURED) {
+				nbcheck++;
+				if ((srv_getinter(&s->check) >= SRV_CHK_INTER_THRES) &&
+				    (!mininter || mininter > srv_getinter(&s->check)))
+					mininter = srv_getinter(&s->check);
+			}
 
-			nbcheck++;
+			if (s->agent.state & CHK_ST_CONFIGURED) {
+				nbcheck++;
+				if ((srv_getinter(&s->agent) >= SRV_CHK_INTER_THRES) &&
+				    (!mininter || mininter > srv_getinter(&s->agent)))
+					mininter = srv_getinter(&s->agent);
+			}
 		}
 	}
 
-- 
1.8.1.5

